redis/src/main/scala/zio/redis/ResultBuilder.scala

```diff
@@ -3,17 +3,24 @@ package zio.redis
 import zio.ZIO
 import zio.schema.Schema
 
-trait ResultBuilder[+F[+_]] {
-  import ResultBuilder.NeedsReturnType
+sealed trait ResultBuilder
 
+trait ResultSchemaBuilder extends ResultBuilder {
+  def returning[R: Schema]: ZIO[RedisExecutor, RedisError, R]
+}
+
+trait ResultSchemaBuilder1[F[_]] extends ResultBuilder {
   def returning[R: Schema]: ZIO[RedisExecutor, RedisError, F[R]]
+}
 
-  final def map(f: Nothing => Any)(implicit nrt: NeedsReturnType): ZIO[Any, Nothing, Nothing] = ???
+trait ResultSchemaBuilder2[F[_, _]] extends ResultBuilder {
+  def returning[R1: Schema, R2: Schema]: ZIO[RedisExecutor, RedisError, F[R1, R2]]
+}
 
-  final def flatMap(f: Nothing => Any)(implicit nrt: NeedsReturnType): ZIO[Any, Nothing, Nothing] = ???
+trait ResultSchemaBuilder3[F[_, _, _]] extends ResultBuilder {
+  def returning[R1: Schema, R2: Schema, R3: Schema]: ZIO[RedisExecutor, RedisError, F[R1, R2, R3]]
 }
 
-object ResultBuilder {
-  @annotation.implicitNotFound("Use `returning[A]` to specify method's return type")
-  final abstract class NeedsReturnType
```

Comment:
`NeedsReturnType` together with `map` and `flatMap` removed above is quite important for the overall user experience. That said, please reintroduce them.

Reply:
I don't see/understand the usefulness of it. Can you give me examples how user suppose to use them?

Reply:
Consider the case where you accidentally omit the call to returning inside the for comprehension. By keeping this in, you'll have a neat hint.

Reply:
Thanks for explanation. It looks like a crutch, but i agree it potentially increase user understanding of the error
Fixed